northstar/src/api/codec.rs

```diff
@@ -23,21 +23,34 @@ use std::{
 };
 use task::Context;
 use tokio::io::{self, AsyncRead, AsyncWrite};
-use tokio_util::codec::{Decoder, Encoder};
+use tokio_util::codec::{Decoder, Encoder, FramedParts};
 
 /// Newline delimited json codec for api::Message that on top implementes AsyncRead and Write
 pub struct Framed<T> {
-    framed: tokio_util::codec::Framed<T, Codec>,
+    inner: tokio_util::codec::Framed<T, Codec>,
 }
 
+impl<T> Framed<T> {
+    /// Consumes the Framed, returning its underlying I/O stream, the buffer with unprocessed data, and the codec.
+    pub fn into_parts(self) -> FramedParts<T, Codec> {
+        self.inner.into_parts()
+    }
+
+    /// Consumes the Framed, returning its underlying I/O stream.
+    pub fn into_inner(self) -> T {
+        self.inner.into_inner()
+    }
+}
+
```

Comment:
I couldn't understand what `into_parts()` does so i tried to look for where it is used but it seems it is not used? The same seems to happen with `into_inner()`. I'm probably missing something, what is this `impl` block for?

Reply:
We wrap the tokio_util::codec::Framed in an struct to make to fix the codec type. This hides the API from `tokio_util::codec::Framed` (`into_parts` and `into_inner`) which is needed if the raw transport shall be obtained back from the struct.

Reply:
Understood, it is just that those methods seem to not be currently used in the project.